src/GafferOSL/OSLShader.cpp

```diff
@@ -84,9 +104,20 @@ struct ShadingEngineCacheGetterKey
 	{
 	}
 
-	ShadingEngineCacheGetterKey( const OSLShader *s )
-		:	shader( s ), hash( s->attributesHash() )
+	ShadingEngineCacheGetterKey( const OSLShader *s, const CompoundObject *targetA )
+		:	shader( s ), targetAttributes( targetA )
 	{
+		hash = s->attributesHash();
+		if( targetAttributes )
+		{
+			
+			ConstCompoundObjectPtr attributes = shader->attributes();
+			const ShaderNetwork *network = findOSLShaderNetwork( attributes.get() );
+			if( network && network->size() )
```

Comment:
Why do we care about size? Won't `ShaderNetwork` deal with that?

Reply:
Fixed in 3c3d696